src/redux/selectors/index.js

```diff
@@ -221,77 +221,56 @@ const selectedCollectionResourceIdsSelector = createSelector(
   (collections, selectedCollection) => collections[selectedCollection]?.resourceIds,
 );
 
-const isResourceInDateRange = (resource, dateRangeStart, dateRangeEnd) => {
-  const { timelineDate } = resource;
-  if (!timelineDate) {
-    console.info('record does not have an timelineDate, resourceId: ', resource.id); // eslint-disable-line no-console
-    return false;
-  }
-  if (!dateRangeStart || !dateRangeEnd) { return true; }
-  return (
-    isWithinInterval(
-      timelineDate, { start: startOfDay(dateRangeStart), end: endOfDay(dateRangeEnd) },
-    )
-  );
-};
-
 const filteredResourceTypesSelector = createSelector(
   [
     resourceTypeFiltersSelector,
     resourceIdsGroupedByTypeSelector,
-    selectedResourceTypeSelector,
     selectedCollectionResourceIdsSelector,
-    dateRangeFilterFiltersSelector,
-    resourcesSelector,
-    timelinePropsSelector,
+    timelineItemsInRangeSelector,
   ],
   (
     resourceTypeFilter,
     resourceIdsGroupedByType,
-    selectedResourceType,
     selectedCollectionResourceIdsObjects,
-    dateRangeFilterFilters,
-    resources,
-    timelineProps,
+    timelineItemsInRange,
   ) => {
     if (!selectedCollectionResourceIdsObjects) {
       return {};
     }
-    const { minimumDate, maximumDate } = timelineProps;
-    const { dateRangeStart, dateRangeEnd } = dateRangeFilterFilters;
-    const activeDateStart = dateRangeStart || minimumDate;
-    const activeDateEnd = dateRangeEnd || maximumDate;
     const selectedCollectionResourceIds = Object.keys(selectedCollectionResourceIdsObjects);
     return Object.keys(resourceTypeFilter).reduce((acc, resourceType) => {
       if (resourceTypeFilter[resourceType]) {
         acc[resourceType] = {};
-        acc[resourceType].selected = false;
-        if (selectedResourceType === resourceType) {
-          acc[resourceType].selected = true;
-        }
+
         Object.entries(resourceIdsGroupedByType[resourceType]).forEach(([subType, resourceIds]) => {
+          const subTypeViewModel = {};
+
+          const subTypeResourceIds = Array.from(resourceIds);
+          subTypeViewModel.resourceIds = subTypeResourceIds;
+          subTypeViewModel.count = subTypeResourceIds.length;
+
+          const dateFilteredResourceIds = timelineItemsInRange
+            .filter((timelineItem) => timelineItem.subType === subType)
+            .reverse()
+            .reduce((array, item) => {
+              array.push(item.id);
+              return array;
+            }, []);
```

Comment:
This is going to execute `n` times for the number of filtered subTypes.

Maybe the primary/outer iteration should be over `timelineItemsInRange`
* it already has type filter applied
* it's already sorted by date (if date ascending is the only applicable filter)

Reply:
I looked at this a few different ways. Since I still need the `resourceType` key, I think this is the correct structure. If you have other ideas, I would welcome a refactor in another pr.